### DISCUSSION TITLE
brew update results in "duck" not launching

### DISCUSSION THREAD

**Discussion by Lastat77** [Homebrew/discussions#2298] [Everyday usage] [answered]
On two separate computer (MacOS 10.14.6 Mojave) I have used "duck" via install from brew for years.

On both systems it was at latest version. Then I ran "brew update". And in the log I see the following:

[
==> Reinstalling 1 broken dependent from source:
duck
==> Downloading https://github.com/java-native-access/jna/archive/refs/tags/5.8.0.tar.gz
==> Downloading from https://codeload.github.com/java-native-access/jna/tar.gz/refs/tags/5.8.0
==> Downloading https://github.com/apple/openjdk/archive/refs/tags/iTunesOpenJDK-1014.0.2.12.1.tar.gz
==> Downloading from https://codeload.github.com/apple/openjdk/tar.gz/refs/tags/iTunesOpenJDK-1014.0.2.12.1
==> Downloading https://dist.duck.sh/duck-src-7.10.2.35432.tar.gz
==> Reinstalling duck 
]

Now duck will not run on either computer.

Thoughts?

**Comment by SMillerDev** [comment]
Well, what do you see when you try to run it? What does `brew doctor` and `brew config` say?

**Reply by Lastat77** [reply]
If I watch Activity Monitor I can see "duck" appear and disappear under a second. Also "duck --help" or any other duck command run from command line does nothing just get a new line.

brew doctor reports nothing related. It had said xcode cli was wrong version...but it has said that before with no issues.

brew config simply puts out a bunch of environment attributes that are all correct.

Also, all my other brew installed apps seem to be fine. It is just duck. But duck version is from Aug 25th (latest) and has not changed. It seems there might be a dependency thing going wrong(?).

**Reply by Lastat77** [reply]
On one computer, I uninstalled brew with command line from brew.sh. I then deleted what was left around. I then reinstalled brew, ran update and upgrade for whatever .. and then installed duck again. It does not work still.

**Comment by cho-m** [comment, marked as answer]
The `Reinstalling 1 broken dependent from source` probably happened due to [`libffi` update](https://github.com/Homebrew/homebrew-core/pull/80227), which had a library version change (`libffi.7 -> libffi.8`).

Not sure why this wasn't detected in that PR, or in any followup PRs.

However, I don't know why rebuilding-from-source has broken `duck`. It could be incompatibility with newer `libffi`. We can try revision bumping `duck` in separate PR to try to rebuild bottle and run test on CI to see if it can reproduce this.

**EDIT:** Created PR to see what happens on CI: https://github.com/Homebrew/homebrew-core/pull/87337

**Reply by Lastat77** [reply]
I appreciate the help. I admit, I don't really know what this all means... but the attention is appreciated.

Is there an action item on my part?

**Reply by cho-m** [reply]
Not right now. The linked pull request (Homebrew/homebrew-core#87337) should either:
- prepare new "bottles" that will be available to `brew upgrade ...` (hopefully fixing issue), or
- show some error message that may help figure out what is going wrong.

**Reply by Lastat77** [reply]
I just ran "brew upgrade" and I see it in the logs updating/upgrading duck.

I just ran some duck scripts and all seems well.

Thank you.

**Reply by cho-m** [reply]
**EDIT:** Didn't see your reply before posting comment. Thanks for confirming it works.

@Lastat77 New bottle should be available after running `brew update`. Please give it a try.

At least `duck --help` works for me on Big Sur:
```console
❯ duck --help
Usage:duck [options...]


URLs must be fully qualified. Paths can either denote a remote file (ftps://user@example.net/resource) or folder (ftps://user@example.net/directory/) with a trailing slash. You can reference files
relative to your home directory with /~ (ftps://user@example.net/~/).
Default protocols
                                     WebDAV (HTTP) dav://<hostname>/<folder>/<file>
                                    WebDAV (HTTPS) davs://<hostname>/<folder>/<file>
                      FTP (File Transfer Protocol) ftp://<hostname>/<folder>/<file>
                       FTP-SSL (Explicit AUTH TLS) ftps://<hostname>/<folder>/<file>
                 SFTP (SSH File Transfer Protocol) sftp://<hostname>/<folder>/<file>
```

**Reply by Lastat77** [reply]
All good. I super appreciate it.
